### PR TITLE
TYP: Create DeepChainMap from ChainMap.new_child in scope.py

### PR DESCRIPTION
**Pull request by kashifkhan**
xref #37715

scope is created as a DeepChainMap but on lines 139, 145, 262 call to new_child was returning ChainMap. Created a new DeepChainMap from the resulting ChainMap.

### CODE REVIEW

File: pandas/core/computation/scope.py [hunk 1]
```diff
@@ -134,17 +134,9 @@ def __init__(
             # scope when we align terms (alignment accesses the underlying
             # numpy array of pandas objects)
 
-            # error: Incompatible types in assignment (expression has type
-            # "ChainMap[str, Any]", variable has type "DeepChainMap[str, Any]")
-            self.scope = self.scope.new_child(  # type: ignore[assignment]
-                (global_dict or frame.f_globals).copy()
-            )
+            self.scope = DeepChainMap(self.scope.new_child( (global_dict or frame.f_globals).copy()))
```
Comment (jbrockmendel) [inline pandas/core/computation/scope.py:137]: can you do this on two lines, so

```
foo = self.scope.whatever
self.scope = DeepChainMap(foo)
```

Comment (kashifkhan) [reply pandas/core/computation/scope.py]: > 

sure, fixed in subsequent commits

**Comment by jreback** [comment]
@kashifkhan seems to be failing precommit; you can run this locally (and fix up)

**Comment by jbrockmendel** [comment]
CI should be fixed if you merge master

**Comment by kashifkhan** [comment]
> CI should be fixed if you merge master

Sure Ill take care of this tomorrow !!

**Comment by jreback** [comment]
thanks @kashifkhan

**Pull request merged**